pkg/proxy/apis/kubeproxyconfig/validation/validation.go

```diff
@@ -147,9 +148,17 @@ func validateProxyMode(mode kubeproxyconfig.ProxyMode, fldPath *field.Path) fiel
 	case kubeproxyconfig.ProxyModeIPTables:
 	case kubeproxyconfig.ProxyModeIPVS:
 	case "":
+	case kubeproxyconfig.ProxyModeKernelspace:
+		if runtime.GOOS != "windows" {
+			errMsg := fmt.Sprintf("%s is only supported on Windows", string(kubeproxyconfig.ProxyModeKernelspace))
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("ProxyMode"), string(mode), errMsg))
+		}
 	default:
 		modes := []string{string(kubeproxyconfig.ProxyModeUserspace), string(kubeproxyconfig.ProxyModeIPTables), string(kubeproxyconfig.ProxyModeIPVS)}
-		errMsg := fmt.Sprintf("must be %s or blank (blank means the best-available proxy (currently iptables)", strings.Join(modes, ","))
+		if runtime.GOOS == "windows" {
```

Comment:
This still isn't right - ipvs and iptables are linux only.  This should just be broken to two different functions.

```
if runtime.GOOS == "windows" {
  return validateProxyModeWindows(mode, fldPath)
} else {
   return validateProxyModeLinux(mode, fldPath)
}
```

wherein each function does just what is allowed for that OS